### PR TITLE
[Feature][Zeta][Core] Support output log file of job

### PR DESCRIPTION
**Pull request by hailin0**
### Purpose of this pull request

[Feature][Zeta][Core] Support output log file of job

ref
https://logging.apache.org/log4j/2.x/manual/appenders/delegating.html#RoutingAppender

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?

Added

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  2. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
  3. Add ci label in [label-scope-conf](https://github.com/apache/seatunnel/blob/dev/.github/workflows/labeler/label-scope-conf.yml)
  4. Add e2e testcase in [seatunnel-e2e](https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/)
  5. Update connector [plugin_config](https://github.com/apache/seatunnel/blob/dev/config/plugin_config)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by hailin0** [comment]
subtask of #7491

**Comment by hailin0** [comment]
cc @arshadmohammad

**Comment by hailin0** [comment]
local testcase

image

**Comment by dailai** [comment]
I think we should add a doc to introduce this feature. It's a great job.

**Comment by hailin0** [comment]
> I think we should add a doc to introduce this feature. It's a great job.

Completed by this subtasks  #7716

**Comment by Hisoka-X** [comment]
Could you add new doc part of logging? Just like flink https://nightlies.apache.org/flink/flink-docs-release-1.20/docs/deployment/advanced/logging/

We should share to users how to configure different log output.

File: config/log4j2.properties [hunk 1]
```diff
@@ -38,7 +38,10 @@ logger.debezium.level=WARN
 #rootLogger.appenderRef.consoleStderr.ref = consoleStderrAppender
 ############################ log output to console #############################
 ############################ log output to file    #############################
-rootLogger.appenderRef.file.ref = fileAppender
+# hybrid log appender
```
Comment (Hisoka-X) [inline config/log4j2.properties]: Can we keep the old behavior? Let user to choose open `routingAppender` or not?

File: config/log4j2.properties [hunk 2]
```diff
@@ -61,13 +64,29 @@ appender.consoleStderr.filter.acceptGteWarn.level = WARN
...
+appender.routing.route.job.appender.layout.pattern = %d{yyyy-MM-dd HH:mm:ss,SSS} %-5p [%-30.30c{1.}] [%t] - %m%n
+
 appender.file.name = fileAppender
 appender.file.type = RollingFile
 appender.file.fileName = ${file_path}/${file_name}.log
 appender.file.filePattern = ${file_path}/${file_name}.log.%d{yyyy-MM-dd}-%i
 appender.file.append = true
 appender.file.layout.type = PatternLayout
-appender.file.layout.pattern = [%X{ST-JID}] %d{yyyy-MM-dd HH:mm:ss,SSS} %-5p [%-30.30c{1.}] [%t] - %m%n
+appender.file.layout.pattern = %d{yyyy-MM-dd HH:mm:ss,SSS} %-5p [%-30.30c{1.}] [%t] - %m%n
```
Comment (Hisoka-X) [inline config/log4j2.properties]: If we do not supported `[%X{ST-JID}]`. We should make sure this PR be merged before 2.3.8 release.

Comment (hailin0) [reply config/log4j2.properties]: Updated docs

**Comment by hailin0** [comment]
> Could you add new doc part of logging? Just like flink https://nightlies.apache.org/flink/flink-docs-release-1.20/docs/deployment/advanced/logging/
> 
> We should share to users how to configure different log output.

added

File: config/log4j2.properties (hunk 1 above)
Comment (hailin0) [reply config/log4j2.properties]: Better to enable by default

File: docs/en/seatunnel-engine/logging.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
+---
+sidebar_position: 14
+---
+
+# How to use logging
```
Comment (Hisoka-X) [inline docs/en/seatunnel-engine/logging.md]:
```suggestion
# Logging
```

**Comment by arshadmohammad** [comment]
Thank you, @hailin0, for working on this feature.
Could you please confirm if an API has been added for seatunnel-web to interact with seatunnel to retrieve logs?

**Comment by hailin0** [comment]
> Thank you, @hailin0, for working on this feature. Could you please confirm if an API has been added for seatunnel-web to interact with seatunnel to retrieve logs?

@arshadmohammad 
You can create an issue to describe your feature request

**Review by Hisoka-X — APPROVED** [review]
LGTM!!!

**Comment by arshadmohammad** [comment]
> @arshadmohammad You can create an issue to describe your feature request
Sure, I will create

**Comment by arshadmohammad** [comment]
Created https://github.com/apache/seatunnel/issues/7724

**Review by Carl-Zhou-CN — APPROVED** [review]
+1

**Pull request merged**